filesystem/amlogic_emmc_partitions.ksy

```diff
@@ -0,0 +1,72 @@
+meta:
+  id: amlogic_emmc_partitions
+  title: Amlogic proprietary eMMC partition table
+  license: CC0-1.0
+  ks-version: 0.9
+  bit-endian: le
+  encoding: UTF-8
+  endian: le
+
+doc: |
+  This is an unnamed and undocumented partition table format implemented by
+  the bootloader and kernel that Amlogic provides for their Linux SoCs (Meson
+  series at least, and probably others). They appear to use this rather than GPT,
+  the industry standard, because their BootROM loads and executes the next stage
+  loader from offset 512 (0x200) on the eMMC, which is exactly where the GPT
+  header would have to start. So instead of changing their BootROM, Amlogic
+  devised this partition table, which lives at an offset of 36MiB (0x240_0000)
+  on the eMMC and so doesn't conflict. This parser expects as input just the
+  partition table from that offset. The maximum number of partitions in a table
+  is 32, which corresponds to a maximum table size of 1304 bytes (0x518).
+  
+doc-ref:
+  - http://aml-code.amlogic.com/kernel/common.git/tree/include/linux/mmc/emmc_partitions.h?id=18a4a87072ababf76ea08c8539e939b5b8a440ef
+  - http://aml-code.amlogic.com/kernel/common.git/tree/drivers/amlogic/mmc/emmc_partitions.c?id=18a4a87072ababf76ea08c8539e939b5b8a440ef
+
+seq:
+  - id: magic
+    contents: ["MPT", 0]
+  - id: version
+    size: 12
+    type: strz
+  - id: num_partitions
+    -orig-id: part_num
+    type: s4
+    valid:
+      min: 1
+      max: 32
+  - id: checksum
+    type: u4
+    doc: |
+      Unsigned modulo 32 sum of all the bytes in the "partitions" element below,
+      interpreted as a sequence of 32-byte little-endian unsigned integers.
+  - id: partitions
+    type: partition
+    repeat: expr
+    repeat-expr: num_partitions
+
+types:
+  partition:
+    seq:
+      - id: name
+        size: 16
+        type: strz
+      - id: size
+        type: u8
+      - id: offset
+        type: u8
+        doc: |
+          The start of the partition relative to the start of the eMMC, in bytes
```

Comment:
Since we know size and position, it may be better to ejcode this info into an instance rather than have it as a free-form description

Reply:
As currently written, the spec is designed to take just the partition table rather than an entire disk image. As such, offsets into the file don't match offsets into eMMC, meaning any partition that exists before the table location would have to have a negative offset.

The reason I chose to do it this way is because the table lives 36MiB into flash, so requiring an eMMC image as input would require parsing a massive file even if all you care about is the partition metadata.